addons/onboarding/models/onboarding_step.py

```diff
@@ -36,49 +36,88 @@ class OnboardingStep(models.Model):
         'onboarding.progress.step', 'step_id', string='Onboarding Progress Step Records', readonly=True,
         help='All related Onboarding Progress Step Records (across companies)')
 
+    is_per_company = fields.Boolean('Is per company', default=True)
     sequence = fields.Integer(default=10)
 
     @api.depends_context('company')
-    @api.depends('progress_ids', 'progress_ids.step_state')
+    @api.depends('progress_ids', 'progress_ids.step_state', 'is_per_company')
     def _compute_current_progress(self):
```

Comment:
> is it for triggering the update after changing the per company field?

Yes. And it will change the value of `current_progress_step_id` and `current_progress_step`.